flocker/node/agents/test/test_blockdevice.py

```diff
@@ -3395,23 +3478,99 @@ class CreateBlockDeviceDatasetInterfaceTests(
     """
 
 
-class CreateBlockDeviceDatasetImplementationTests(SynchronousTestCase):
+class CreateBlockDeviceDatasetImplementationMixin(object):
     """
-    ``CreateBlockDeviceDataset`` implementation tests.
+    Utility Mixin for ``CreateBlockDeviceDataset`` implementation tests.
     """
-    def setUp(self):
-        self.api = loopbackblockdeviceapi_for_test(
-            self,
-            allocation_unit=LOOPBACK_ALLOCATION_UNIT
+
+    def _create_blockdevice_dataset(self, dataset_id, maximum_size,
+                                    metadata=pmap({})):
+        """
+        Call ``CreateBlockDeviceDataset.run`` with a ``BlockDeviceDeployer``.
+
+        :param UUID dataset_id: The uuid4 identifier for the dataset which will
+            be created.
+        :param int maximum_size: The size, in bytes, of the dataset which will
+            be created.
+        :param pmap(unicode, unicode) metadata: The metadata for the dataset.
+
+        :returns: A 3-tuple of:
+            * ``BlockDeviceVolume`` created by the run operation
+            * The ``FilePath`` of the device where the volume is attached.
+            * The ``FilePath`` where the volume is expected to be mounted.
+        """
+        expected_mountpoint = self.mountroot.child(
+            unicode(dataset_id).encode("ascii")
         )
-        self.mountroot = mountroot_for_test(self)
-        self.deployer = BlockDeviceDeployer(
-            node_uuid=uuid4(),
-            hostname=u"192.0.2.10",
-            block_device_api=self.api,
-            mountroot=self.mountroot
+
+        dataset = Dataset(
+            dataset_id=unicode(dataset_id),
+            maximum_size=maximum_size,
+            metadata=metadata
+        )
+
+        change = CreateBlockDeviceDataset(
+            dataset=dataset, mountpoint=expected_mountpoint
         )
 
+        run_state_change(change, self.deployer)
```

Comment:
This supposes that the state change is implemented as a synchronous operation - which `CreateBlockDeviceDataset` happens to be, but isn't necessarily the case.  I see this is just refactored code, though, so pre-existing problem, not something to fix right now - just thought I'd mention it.